### PR TITLE
[orange-math] new port

### PR DESCRIPTION
**Pull request by luadebug**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/omath/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE
+    "${CURRENT_PACKAGES_DIR}/debug/include"
+)
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```
Comment (Mengna-Li) [inline ports/omath/portfile.cmake:35]: This is unnecessary, the heuristically generated usage is the same as yours. Please remove your usage file.

File: ports/omath/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,36 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO orange-cpp/omath
+    REF v${VERSION}
```
Comment (Mengna-Li) [inline ports/omath/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/omath/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,36 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO orange-cpp/omath
+    REF v${VERSION}
+    SHA512 467b1abbdf5b9a7f49ed50824eaa4641f05d6088e84f40320b5c82a1bdbf685cc8d0f0a4f4ab6be49e3a8ed13103ee3e808dde3b556a00742f7b53c519c183e3
+    HEAD_REF master
+)
+
+file(REMOVE_RECURSE "${SOURCE_PATH}/extlibs" "${SOURCE_PATH}/tests")
```
Comment (Mengna-Li) [inline ports/omath/portfile.cmake:9]: Can you explain this line?

**Review by Mengna-Li — CHANGES_REQUESTED** [review]

**Comment by Mengna-Li** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/omath/portfile.cmake (hunk 3 above)
Comment (luadebug) [reply ports/omath/portfile.cmake]: Unnecessary as well, hence tests are disabled.

**Review by Mengna-Li — APPROVED** [review]

**Comment by Mengna-Li** [comment]
Usage tested pass on `x64-windows`.

**Comment by JavierMatosD** [comment]
Hi @luadebug, thank you for the contribution! 

Would you mind changing the name of the port to something like `orange-math` or something of the sort? I didn't see this project on repology and the `omath` library didn't come up when I googled it.

**Comment by luadebug** [comment]
Sure thanks for reviewing and help.

**Pull request merged**